lib/togglefy/analytics.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module Togglefy
+  class Analytics
+    def initialize(identifier = nil)
+      @identifier = identifier
+      @feature = fetch_feature
+    end
+
+    def track
+      return [] unless @feature
+
+      build_tracking_data
+    end
+
+    private
+
+    def fetch_feature
+      Togglefy.feature(@identifier) || nil
+    end
+
+    def build_tracking_data
+      assignables.filter_map do |assignable|
+        assignable_class = safe_constantize(assignable)
+        next unless assignable_class
+
+        assignable_data = build_assignables_data(assignable_class)
+        assignments_data = build_assignments_data
+
+        {
+          assignable: assignable,
+          total: assignable_data[:total_count],
+          enabled_count: assignable_data[:enabled_count],
+          disabled_count: assignable_data[:disabled_count],
+          percentage_enabled: assignable_data[:percentage_enabled],
+          percentage_disabled: assignable_data[:percentage_disabled],
+          first_created: assignments_data[:first_created],
+          last_created: assignments_data[:last_created],
+          past7: assignments_data[:past7],
+          past14: assignments_data[:past14],
+          past30: assignments_data[:past30]
+        }
```

Comment:
Both `assignable_data` and `assignments_data` already returns an array, right?

So maybe, instead of building this return manually, we could just create a hash with the `assignable` and merge the other two?

```suggestion
        assignable_data = build_assignables_data(assignable_class)
        assignments_data = build_assignments_data

        [{ assignable: assignable }, assignable_data, assignments_data].reduce(:merge)
```

OR

```suggestion
        assignable_data = build_assignables_data(assignable_class)
        assignments_data = build_assignments_data

        { assignable: assignable }.merge(assignable_data).merge(assignments_data)
```

Reply:
Nice!